ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -51,10 +51,9 @@ private[server] object ServerHelpers {
     Response(Status.InternalServerError).putHeaders(org.http4s.headers.`Content-Length`.zero)
 
   def server[F[_]](
-      bindAddress: InetSocketAddress,
+      serverResource: Resource[F, (SocketAddress[IpAddress], Stream[F, Socket[F]])],
```

Comment:
Did fs2 change the signature here ? IIRC it was `Stream[F, Resource[F, Socket[F]]]`

Reply:
Yes.  @yanns called that out on his PR merged into this.  I think his misgivings were right, but I'm flying around too fast working on other dependencies.  I think our problems start here, though.

Reply:
So I'm not sure if there is a relationship between this and the server _not_ shutting down yet, but this signature isn't going to work for graceful shutdowns. By allocating the sockets on the server stream, there's no way we can interrupt that stream while letting handler streams proceed for some more time.

Reply:
Honestly, I feel like these signatures might be bad news:
```scala
def server(
        address: Option[Host],
        port: Option[Port],
        options: List[SocketOption]
    ): Stream[F, Socket[F]]

def serverResource(
        address: Option[Host],
        port: Option[Port],
        options: List[SocketOption]
    ): Resource[F, (SocketAddress[IpAddress], Stream[F, Socket[F]])]
```
Think about it: the server is a `Stream` loop that keeps on allocating a new `Socket` resource and moves onto accepting the next one. It's going to cause a memory leak if it runs long enough because the socket finalizers are piling up!

Reply:
Okay, maybe the finalizers won't stack up like that: https://github.com/typelevel/fs2/blob/main/io/src/main/scala/fs2/io/net/SocketGroup.scala#L181

The `++` before the next accept means that the current scope introduced by `Stream.resource` is cleaned up